xrpl4j-model/src/main/java/org/xrpl/xrpl4j/model/transactions/Wrappers.java

```diff
@@ -118,12 +128,23 @@ public static XrpCurrencyAmount ofDrops(UnsignedLong drops) {
      * @return An {@link XrpCurrencyAmount} of the amount of drops in {@code amount}.
      */
     public static XrpCurrencyAmount ofXrp(BigDecimal amount) {
-      return ofDrops(amount.scaleByPowerOfTen(6).toBigIntegerExact().longValue());
+      if (FluentCompareTo.is(amount).notEqualTo(BigDecimal.ZERO)) {
+        Preconditions.checkArgument(FluentCompareTo.is(amount).greaterThanEqualTo(SMALLEST_XRP));
+      }
+      return ofDrops(UnsignedLong.valueOf(amount.scaleByPowerOfTen(6).toBigIntegerExact()));
     }
 
-    @Override
-    public String toString() {
-      return this.value().toString();
+    /**
+     * Convert the supplied amount of XRP drops into a decimal value representing a value denominated in whole XRP
+     * units.
+     *
+     * @param xrpCurrencyAmount A {@link XrpCurrencyAmount} representing an amount of XRP drops.
+     *
+     * @return A {@link BigDecimal} representing a value denominated in whole XRP units.
+     */
+    static BigDecimal toXrp(final XrpCurrencyAmount xrpCurrencyAmount) {
```

Comment:
This would be more obvious and convenient to use as a default method on XrpCurrency. So I could do something like `payment.amountAsXrp().toBigDecimal()` rather than `XrpCurrencyAmount.toXrp(payment.amountAsXrp())`

Reply:
Agree, great suggestion! Changed [here](https://github.com/XRPLF/xrpl4j/pull/68/commits/195c7a87fd2bbb5c31e54353fb9ca56674c4415d).

Reply:
👍